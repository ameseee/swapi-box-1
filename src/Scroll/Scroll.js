import React from 'react';
import PropTypes from 'prop-types';

const Scroll = ( {scrollData} ) => {
  return (
    <div className="scroll-text">
      <div className="scroll-cont">
        <p className="scroll-p">
          {scrollData[6].opening_crawl}          
        </p>
      </div>
    </div>
  );
};

Scroll.propTypes = {
  scrollData: PropTypes.string
};

export default Scroll;
