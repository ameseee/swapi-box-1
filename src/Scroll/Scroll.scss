.scroll-text {
  overflow: hidden;
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.scroll-cont {
  display: flex;
  justify-content: center;
  position: relative;
  height: 300px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 200%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
}

.scroll-p {
  position: relative;
  top: 9999px;
  transform-origin: 50% 100%;
  animation: crawl 120s linear;
}

.scroll-p  > .title {
  font-size: 90%;
  text-align: center;
}

.scroll-p  > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: 400px;
    transform: rotateX(20deg)  translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
