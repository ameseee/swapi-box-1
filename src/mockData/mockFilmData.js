const mockFilmData = [
  {
    "title": "A New Hope",
    "episode_id": 4,
    "opening_crawl": "It is a period of civil war.",
    "director": "George Lucas",
    "producer": "Gary Kurtz, Rick McCallum",
    "release_date": "1977-05-25"
  },
  {
    "title": "Attack of the Clones",
    "episode_id": 2,
    "opening_crawl": "There is unrest in the Galactic\r\nSenate.",
    "director": "George Lucas",
    "producer": "Rick McCallum",
    "release_date": "2002-05-16"
  },
  {
    "title": "The Phantom Menace",
    "episode_id": 1,
    "opening_crawl": "Turmoil has engulfed the\r\nGalactic Republic.",
    "director": "George Lucas",
    "producer": "Rick McCallum",
    "release_date": "1999-05-19"
  }];

export default mockFilmData;
