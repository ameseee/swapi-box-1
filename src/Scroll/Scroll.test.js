import React from 'react';
import Scroll from './Scroll';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockFilmData from '../mockData/mockFilmData';

describe('Scroll', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const wrapper = shallow(<Scroll scrollData={[mockFilmData[0]]} />);
  const scrollDiv = wrapper.find('.scroll-text');

  it('should exist', () => {
    expect(scrollDiv.length).toEqual(1);
  });

  it('should have inner text corresponding to prop', () => {
    expect(scrollDiv.text()).toEqual(
      mockFilmData[0].opening_crawl +
      mockFilmData[0].title +
      mockFilmData[0].release_date
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
