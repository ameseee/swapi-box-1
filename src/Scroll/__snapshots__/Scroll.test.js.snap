// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll
    scrollData={
        Array [
            Object {
              "director": "George Lucas",
              "episode_id": 4,
              "opening_crawl": "It is a period of civil war.
        Rebel spaceships, striking
        from a hidden base, have won
        their first victory against
        the evil Galactic Empire.
        
        During the battle, Rebel
        spies managed to steal secret
        plans to the Empire's
        ultimate weapon, the DEATH
        STAR, an armored space
        station with enough power
        to destroy an entire planet.
        
        Pursued by the Empire's
        sinister agents, Princess
        Leia races home aboard her
        starship, custodian of the
        stolen plans that can save her
        people and restore
        freedom to the galaxy....",
              "producer": "Gary Kurtz, Rick McCallum",
              "release_date": "1977-05-25",
              "title": "A New Hope",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ticker-wrap"
>
        <p
                className="ticker"
        >
                It is a period of civil war.
                Rebel spaceships, striking
                from a hidden base, have won
                their first victory against
                the evil Galactic Empire.
                
                During the battle, Rebel
                spies managed to steal secret
                plans to the Empire's
                ultimate weapon, the DEATH
                STAR, an armored space
                station with enough power
                to destroy an entire planet.
                
                Pursued by the Empire's
                sinister agents, Princess
                Leia races home aboard her
                starship, custodian of the
                stolen plans that can save her
                people and restore
                freedom to the galaxy....
                <br />
                <br />
                A New Hope
                <br />
                <br />
                1977-05-25
        </p>
</div>,
      "className": "scroll-text",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="ticker"
>
          It is a period of civil war.
          Rebel spaceships, striking
          from a hidden base, have won
          their first victory against
          the evil Galactic Empire.
          
          During the battle, Rebel
          spies managed to steal secret
          plans to the Empire's
          ultimate weapon, the DEATH
          STAR, an armored space
          station with enough power
          to destroy an entire planet.
          
          Pursued by the Empire's
          sinister agents, Princess
          Leia races home aboard her
          starship, custodian of the
          stolen plans that can save her
          people and restore
          freedom to the galaxy....
          <br />
          <br />
          A New Hope
          <br />
          <br />
          1977-05-25
</p>,
        "className": "ticker-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
            <br />,
            <br />,
            "A New Hope",
            <br />,
            <br />,
            "1977-05-25",
          ],
          "className": "ticker",
        },
        "ref": null,
        "rendered": Array [
          "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "A New Hope",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "1977-05-25",
        ],
        "type": "p",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ticker-wrap"
>
          <p
                    className="ticker"
          >
                    It is a period of civil war.
                    Rebel spaceships, striking
                    from a hidden base, have won
                    their first victory against
                    the evil Galactic Empire.
                    
                    During the battle, Rebel
                    spies managed to steal secret
                    plans to the Empire's
                    ultimate weapon, the DEATH
                    STAR, an armored space
                    station with enough power
                    to destroy an entire planet.
                    
                    Pursued by the Empire's
                    sinister agents, Princess
                    Leia races home aboard her
                    starship, custodian of the
                    stolen plans that can save her
                    people and restore
                    freedom to the galaxy....
                    <br />
                    <br />
                    A New Hope
                    <br />
                    <br />
                    1977-05-25
          </p>
</div>,
        "className": "scroll-text",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="ticker"
>
            It is a period of civil war.
            Rebel spaceships, striking
            from a hidden base, have won
            their first victory against
            the evil Galactic Empire.
            
            During the battle, Rebel
            spies managed to steal secret
            plans to the Empire's
            ultimate weapon, the DEATH
            STAR, an armored space
            station with enough power
            to destroy an entire planet.
            
            Pursued by the Empire's
            sinister agents, Princess
            Leia races home aboard her
            starship, custodian of the
            stolen plans that can save her
            people and restore
            freedom to the galaxy....
            <br />
            <br />
            A New Hope
            <br />
            <br />
            1977-05-25
</p>,
          "className": "ticker-wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
              <br />,
              <br />,
              "A New Hope",
              <br />,
              <br />,
              "1977-05-25",
            ],
            "className": "ticker",
          },
          "ref": null,
          "rendered": Array [
            "It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy....",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "A New Hope",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "1977-05-25",
          ],
          "type": "p",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
